aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -327,6 +327,36 @@ void gemm(
       transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
 
+void gemm(
+    TransposeType transa, TransposeType transb,
+    int64_t m, int64_t n, int64_t k,
+    const float alpha,
+    const at::BFloat16 *a, int64_t lda,
+    const at::BFloat16 *b, int64_t ldb,
+    const float beta,
+    float *c, int64_t ldc) {
+  internal::normalize_last_dims(transa, transb, m, n, k, &lda, &ldb, &ldc);
+#ifdef MKL_HAS_SBGEMM
+  if (use_blas_gemm(transa, transb, m, n, k, lda, ldb, ldc)) {
+    int m_ = m, n_ = n, k_ = k, lda_ = lda, ldb_ = ldb, ldc_ = ldc;
+    mkl_gemm_bf16bf16f32(transa, transb, m_, n_, k_, alpha, a, lda_, b, ldb_, beta, c, ldc_);
```

Comment:
Can we just use `sbgemm_` directly in `CPUBlas.cpp`, which is already used in the all-`BFloat16` overload?

Reply:
No, I can not directly use `sbgemm_` in `CPUBlas.cpp`, since `sbgemm_` is from OpenBlas and MKL doesn't have a routine with the same name.

But I just added a path with `sbgemm_` in this function as well, so that if PyTorch is built against OpenBlas, it will also receive speedup using the scaled dot product attention optimization (e.g. flash attention).